src/network/message.rs

```diff
@@ -144,6 +161,40 @@ impl RawNetworkMessage {
             NetworkMessage::Alert(_)    => "alert",
         }.to_owned()
     }
+
+    /// Reads stream from a TCP socket and parses first message from it, returing
+    /// the rest of the unparsed buffer for later usage.
+    pub fn from_stream(stream: &mut Read, remaining_part: &mut Vec<u8>,
+                       StreamReaderConfig { iterations, buffer_size }: StreamReaderConfig) -> Result<Self, encode::Error> {
+        println!("Called with {} iterations and {} ubffer size", iterations, buffer_size);
```

Comment:
would be nicer to have a StreamReader that is constructed with a buffer size and retry parameter and this a method of that class. Remaining buffer should not be exposed to the user of this call as it is irrelevant, the user only wants to get full messages all the rest is implementation detail that should  rather be hidden, as it might also change.

Reply:
Completely reworked the StreamReader, put it into a standalone mod/file and refactored its unit tests. Also, because of different workflow for the message parsing have got rid of `iterations` parameter: we have to wait for the network to deliver all packets for the message if we can't parse a single message from the buffered part.